src/common/utxobased/keymanager/keymanager.ts

```diff
@@ -904,10 +906,20 @@ export async function makeTx(args: MakeTxArgs): Promise<MakeTxReturn> {
   }
 
   const targets: utxopicker.Target[] = args.targets.map(target => {
-    const script = addressToScriptPubkey({
+    let script = addressToScriptPubkey({
       address: target.address,
       coin: coin.name
     })
+    if (target.memo != null) {
+      const memoHex = Buffer.from(target.memo, 'utf8').toString('hex')
+      // check if hex string is within 80 bytes
+      if (getNumberOfBytesFromHex(memoHex) > 80) {
+        throw new Error('Memo size exceeds 80 bytes')
+      }
```

Comment:
This should be written as the 'validateMemo' function in the makeCurrencyTools following this typing - https://github.com/EdgeApp/edge-core-js/blob/afbdb5ece2964ea09c29fb28b816ad8d9e726b19/src/types/types.js#L670

Reply:
This is still not resolved, you should put the function on the currencyTools object.